src/main/java/net/pms/dlna/DLNAOrgFlags.java

```diff
@@ -0,0 +1,564 @@
+	 *         is {@code 0}.
+	 */
+	public boolean isInteractive() {
+		return ((high >> 55) & 1) == 1;
+	}
+
+	/**
+	 * Bit 22: tm-b (Background mode flag).
+	 *
+	 * @return {@code true} if the flag is {@code 1}, {@code false} if the flag
+	 *         is {@code 0}.
+	 */
+	public boolean isBackgroundMode() {
+		return ((high >> 54) & 1) == 1;
+	}
+
+	/**
+	 * Bit 21: http-stalling (HTTP Connection Stalling flag).
+	 *
+	 * @return {@code true} if the flag is {@code 1}, {@code false} if the flag
+	 *         is {@code 0}.
+	 */
+	public boolean isHttpConnectionStalling() {
+		return ((high >> 53) & 1) == 1;
+	}
+
+	/**
+	 * Bit 20: dlna-v1.5-flag (DLNA v1.5 versioning flag).
+	 *
+	 * @return {@code true} if the flag is {@code 1}, {@code false} if the flag
+	 *         is {@code 0}.
+	 */
+	public boolean isDLNA15() {
+		return ((high >> 52) & 1) == 1;
+	}
+
+	/**
+	 * Bit 16: LP-flag (Link Protected content flag).
+	 *
+	 * @return {@code true} if the flag is {@code 1}, {@code false} if the flag
+	 *         is {@code 0}.
+	 */
+	public boolean isLinkProtectedContent() {
+		return ((high >> 48) & 1) == 1;
+	}
+
+	/**
+	 * Bit 15: cleartextbyteseek-full flag (Cleartext Byte Full Data Seek flag).
+	 *
+	 * @return {@code true} if the flag is {@code 1}, {@code false} if the flag
+	 *         is {@code 0}.
+	 */
+	public boolean isCleartextByteFullDataSeek() {
+		return ((high >> 47) & 1) == 1;
+	}
+
+	/**
+	 * Bit 14: lop-cleartextbytes flag (Cleartext Limited Data Seek flag).
+	 *
+	 * @return {@code true} if the flag is {@code 1}, {@code false} if the flag
+	 *         is {@code 0}.
+	 */
+	public boolean isCleartextLimitedDataSeek() {
+		return ((high >> 46) & 1) == 1;
+	}
+
+	/**
+	 * Calculates the "effective" flags for this {@link DLNAOrgFlags} instance
+	 * by applying DLNA conditional rules for some flags.
+	 *
+	 * @param mediaType the {@link MediaType} of the media for which this
+	 *            instance applies.
+	 * @return The new {@link DLNAOrgFlags} instance where illegal combinations
+	 *         have been corrected.
+	 */
+	public DLNAOrgFlags getEffectiveFlags(MediaType mediaType) {
+		return DLNAOrgFlags.getEffectiveFlags(this, mediaType);
+	}
+
+	/**
+	 * Gets the parameter string.
+	 *
+	 * @return A string formatted as a DLNA ProtocolInfo parameter or an empty
+	 *         string if the parameter can be omitted.
+	 */
+	public String getParameterString() {
+		DLNAOrgFlags flags = DLNAOrgFlags.getEffectiveFlags(this, MediaType.UNKNOWN);
+		if (flags.high == 0 || (flags.high & ~(3L << 55)) == 0) {
+			return "";
+		}
+		return ";DLNA.ORG_FLAGS=" + String.format("%016x", flags.high) + String.format("%016x", flags.low);
+	}
+
+	/**
+	 * Gets the first 32 bits representing the {@code primary-flags}.
+	 *
+	 * @return The most significant {@link Integer}.
+	 */
+	public int getPrimaryValue() {
+		return (int) (high >> 32);
+	}
+
+	/**
+	 * Gets the first 64 bits of the 128 bits in {@code DLNA.ORG_FLAGS}.
```

Comment:
The largest "normal" integer type in Java is ```long```. Long is 64 bits, while a normal ```int``` is 32 bits. When I decided on how to store these 128 bits I basicly had the option of using 4 ```int```s or 2 ```long```s. Although using 4 ```int```s would have been convenient because only the first is actually in use, I opted to use two ```long```s called "high" and "low". I figured it was "cleaner" dividing the 128 bits in to two than 4, it's "easier" to exchange the information in two variables than in four.